hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataTableValidator.java

```diff
@@ -751,6 +756,7 @@ private void validateFileSlices(
     if (fileSliceListFromMetadataTable.size() != fileSliceListFromFS.size()) {
       mismatch = true;
     } else if (!fileSliceListFromMetadataTable.equals(fileSliceListFromFS)) {
+      Map<String, Set<String>> committedFilesMap = new HashMap<>();
```

Comment:
Do we need to init this committedFilesMapand passing it into areFileSliceCommittedLogFilesMatching(xxx) or we can just init this map in this func `areFileSliceCommittedLogFilesMatching`?

Reply:
`committedFilesMap` serves as an in-memory cache for the set of committed files of each commit.  We don't want to deserialize the same commit metadata repeatedly across multiple calls of `areFileSliceCommittedLogFilesMatching`.